### DISCUSSION TITLE
Websocket API

### DISCUSSION THREAD

**Discussion by JTF4** [josephdadams/TallyArbiter#245] [Ideas]
@MatteoGheza I figured that I would move this here instead of in the blackmagic feature request.

After looking at it, I think that websockets would be a better protocol than rest. It seems like it is a lighter weight protocol than socket.io, so It may have some benefits with battery powered listeners. 

I still think we need a basic rest API, but it does not have to handle reassigns and all that stuff.

**Comment by hrueger** [comment]
I don't think REST is suited for Tally Data at all, because it is hard to push events from the server without a client asking for it. The only way I know of on how to achieve this with REST is with Server Sent Events, which turned out to not be that reliable (at least for me). I had all kinds of issues with buffering, caches and so on.

Why not make a simple TCP / UDP Protocol witout Websockets?

Btw: as fas as I know, socket.io uses websockets internally with polling as a fallback.

**Reply by JTF4** [reply]
Yea socket.io is based on websockets, but that polling makes its packets bigger and has a much higher demand on the CPU of the device that it's running on. 

I would be a fan of the custom protocol implementation.
